util/movement-types/src/lib.rs

```diff
@@ -50,25 +60,52 @@ impl Transaction {
 		Self { data, sequence_number, id }
 	}
 
-	pub fn id(&self) -> Id {
-		self.id.clone()
+	pub fn id(&self) -> &Id {
+		&self.id
+	}
+
+	pub fn data(&self) -> &Vec<u8> {
```

Comment:
```suggestion
	pub fn data(&self) -> &[u8] {
```

There is never a need to return a reference to an immutable `Vec`.

Reply:
It's more nuanced than that. You should return the borrowed form of data in accessors by reference, so for `Vec<T>` it is `&[T]`, for `String` you return `&str`, etc. The standard owned containers add no useful API to their `Deref` target when they are immutably borrowed.

Reply:
Okay